static/js/typing_data.js

```diff
@@ -64,20 +82,40 @@ export function get_all_pms_typists() {
     return pms_typists;
 }
 
+export function get_stream_typists(stream_id, topic) {
+    return typist_dct.get(JSON.stringify({stream_id, topic})) || [];
+}
+
 // The next functions aren't pure data, but it is easy
 // enough to mock the setTimeout/clearTimeout functions.
-export function clear_inbound_timer(group) {
-    const key = get_key(group);
+function clear_inbound_timer(key) {
     const timer = inbound_timer_dict.get(key);
     if (timer) {
         clearTimeout(timer);
         inbound_timer_dict.set(key, undefined);
     }
 }
 
-export function kickstart_inbound_timer(group, delay, callback) {
+export function clear_pms_inbound_timer(group) {
     const key = get_key(group);
-    clear_inbound_timer(group);
+    clear_inbound_timer(key);
+}
+
+export function clear_streams_inbound_timer(stream_id, topic) {
+    const key = JSON.stringify({stream_id, topic});
+    clear_inbound_timer(key);
+}
+
+export function kickstart_pms_inbound_timer(group, delay, callback) {
+    const key = get_key(group);
+    clear_pms_inbound_timer(group);
+    const timer = setTimeout(callback, delay);
+    inbound_timer_dict.set(key, timer);
+}
+
+export function kickstart_streams_inbound_timer(stream_id, topic, delay, callback) {
+    const key = JSON.stringify({stream_id, topic});
+    clear_streams_inbound_timer(stream_id, topic);
     const timer = setTimeout(callback, delay);
     inbound_timer_dict.set(key, timer);
 }
```

Comment:
I'm not sure you actually need to duplicate these functions -- can't we just have whatever is calling them fetch the key and call `kickstart_inbound_timer(key, delay, callback)`?

Reply:
Yes, agreed, that's a lot of duplication. Changed to fetch the key and call `kickstart_inbound_timer(key, delay, callback)` as suggested.